pyfixest/estimation/literals.py

```diff
@@ -0,0 +1,36 @@
+from typing import Any, Literal, get_args
+
+prediction_type = Literal["response", "link"]
+vcov_type_options = Literal["iid", "hetero", "HC1", "HC2", "HC3"]
+weights_type_options = Literal["aweights", "fweights"]
+fixef_rm_options = Literal["singleton", "none"]
+solver_options = Literal["np.linalg.solve", "np.linalg.lstsq"]
+
+
+def validate_literal_argument(arg: Any, literal: Any) -> None:
```

Comment:
Can we restrict the type of `literal`? I think we only want to call this function on literals?

Reply:
Hi @leostimpfle, thanks for reviewing. 

I can look into this. I tried using Literal as a type, but at least when linting, a specification of the options is required.

Reply:
Found the _LiteralGenericAlias type that seems like a good match.

Reply:
Yep, this looks good to me!